### ISSUE TITLE
changeView is not triggering viewRender when events returned via callback

### ISSUE THREAD

**Issue by cdutson** [fullcalendar/fullcalendar#4142]
Version 3.9.0

TL;DR: When starting in listMonth (or week, or day, doesn't matter), and triggering a changeView on page resize via event listener, the new view is not triggered. This only seems to happen if the eventData is in a callback scenario.

I've tried a lot to figure this one out, but after 3 hours of debugging to make sure it's not some weird hiccup in my code, I'm here to bring up a rather... odd bug.

Basically I have a calendar that loads events from a couple end-points and renders them in a calendar. Works great. I decided to add some code to change the view if the user is in 'mobile' view (768 or smaller). specifically I will force the user into 'listMonth' on mobile, but otherwise they can be in month, agendaWeek, or agendaDay. I have a debounced resize event on the page that checks the window size and decides if it should change the view or not. It works... mostly.

If you start in large mode, and size down and up, it works fine. If, however, you start in mobile view (which forces you into listMonth) and then size up viewRender is not being triggered. If you size back down (to listMonth) and then back up, viewRender is called with 'month' as expected.

This only seems to happen if the events prop is set to a function with a callback; If i feed the array directly, it's fine.

I've baked up a bare-bones example here: https://codepen.io/anon/pen/KoEavN?editors=0011

To reproduce in my example:
- load the page up in a small viewport
- wait 1ms for data to load (listMonth will render fine)
- resize window to a larger size (greater than 768)

Expected: viewRender should console.log with the current view (month)
Actual: the event is not fired

**EDIT**
Upon further review, I've found that the issue seems specific to the month view. if you change month to basicWeek, the error goes away. Also, if you trigger the existing view _and then_ the month view, it works as expected.

**Comment by acerix** [comment]
Thanks for the detailed report.  The problem is that when events are supplied as a function, that function needs to return the events in `callback` before the function exits (ie. syncronously).  It works if I disable your timeout:

https://codepen.io/anon/pen/jzoqZQ?&editors=001

**Comment by cdutson** [comment]
Right, but... that's a problem. This is a simplified example, with the timeout standing in for an ajax-style request. In my production code, I fetch events from a couple sources, process them, and then stitch them together before returning a unified array. Obviously that's a bit hard to demonstrate for this example.

And further to the point, it's only when going from a non-month view (my case is listMonth) to month. If I swap it to any other view (basicWeek, agendaDay, whatever) it works as expected.

**Comment by acerix** [comment]
Thanks for the clarification, you're right that it calling `callback` later is fine.

I noticed that it works if you don't call `onResize()` at the end, so you could just set the initial view using `defaultView` instead of changing the view right after the calendar is loaded.

I'm guessing the issue is related to the second render happening before the timeout returning the events for the first render.

**Comment by cdutson** [comment]
Hmmmmm That does seem to do the trick. I can augment my code to perform with the defaultView. Not sure if this continues to count as an error? I feel like there should probably be some catch for this scenario, though I'm not sure how you'd go about doing that.

**Comment by arshaw** [comment]
this is almost definitely fixed in v5
